pytorch_lightning/metrics/metric.py

```diff
@@ -17,6 +19,16 @@ class Metric(DeviceDtypeModuleMixin, torch.nn.Module, ABC):
     Should be used to implement metrics that
     1. Return multiple Outputs
     2. Handle their own DDP sync
+
+    Metric hooks that can be implemented are:
+        input_convert: pre-forward hook that takes care of input conversion
+        output_convert: post-forward hook that takes care of output convertion
+        ddp_sync: implementation of ddp sync
+        compute: post-ddp sync for additional metric computations
+
+    Call order:
+        input_convert -> forward -> output_convert -> ddp_sync -> compute
```

Comment:
Regarding ddp_sync: Maybe this should just be called `aggregate` and can be used to aggregate different batches on the same node as well? The behaviour shouldn't be different from syncing between nodes/processes, right?

So maybe we can implement a proper aggregation method and don't reduce the outputs during ddp_sync, but just broadcast them and then call `aggregate` on this?